pkgs/initiator/initiator.go

```diff
@@ -212,68 +212,78 @@ func (c *Initiator) initMessageFlowHandling(init *spec.Init, id [24]byte, operat
 	return finalResults, nil
 }
 
-func (c *Initiator) ResignMessageFlowHandling(rMsg *wire.ResignMessage, id [24]byte, operators []*spec.Operator) ([][]byte, error) {
-	dkgResult, errs, err := c.SendResignMsg(id, rMsg, operators)
+func (c *Initiator) ResignMessageFlowHandling(rMsg *wire.ResignMessage, sig *wire.SignatureForHash, id [24]byte, operators []*spec.Operator) ([][]byte, error) {
+	_, _, err := c.SendResignMsg(id, rMsg, operators)
+	if err != nil {
+		return nil, err
+	}
+	resignResult, errs, err := c.SendSignResignMsg(id, sig, operators)
 	if err != nil {
 		return nil, err
 	}
 	// sanity check
-	if err := checkThreshold(dkgResult, errs, operators, operators, len(operators)); err != nil {
+	if err := checkThreshold(resignResult, errs, operators, operators, len(operators)); err != nil {
 		return nil, err
 	}
-	err = verifyMessageSignatures(id, dkgResult, c.VerifyMessageSignature)
+	err = verifyMessageSignatures(id, resignResult, c.VerifyMessageSignature)
 	if err != nil {
 		return nil, err
 	}
 	c.Logger.Info("✅ verified operator response signatures")
 	var results [][]byte
-	for _, res := range dkgResult {
+	for _, res := range resignResult {
 		results = append(results, res)
 	}
 	return results, nil
 }
 
-func (c *Initiator) messageFlowHandlingReshare(id [24]byte, reshareMsg *wire.ReshareMessage) ([][]byte, error) {
-	c.Logger.Info("phase 1: sending reshare message to all operators")
-	allOps, err := utils.JoinSets(reshareMsg.SignedReshare.Reshare.OldOperators, reshareMsg.SignedReshare.Reshare.NewOperators)
+func (c *Initiator) messageFlowHandlingReshare(id [24]byte, reshareMsg *wire.ReshareMessage, sig *wire.SignatureForHash) ([][]byte, error) {
+	c.Logger.Info("phase 1: sending unsigned reshare message to all operators")
+	allOps, err := utils.JoinSets(reshareMsg.Reshare.OldOperators, reshareMsg.Reshare.NewOperators)
 	if err != nil {
 		return nil, err
 	}
 	var errs map[uint64]error
-	exchangeMsgs, errs, err := c.SendReshareMsg(id, reshareMsg, allOps)
+	_, _, err = c.SendReshareMsg(id, reshareMsg, allOps)
+	if err != nil {
+		return nil, err
+	}
+	c.Logger.Info("phase 1: ✅ unsigned reshare message sent to all operators")
+	c.Logger.Info("phase 2: sending reshare message signature to all operators")
+	exchangeMsgs, errs, err := c.SendSignReshareMsg(id, sig, allOps)
```

Comment:
The major change of this PR is to break message and signature into two phases, they are now two separate functions on the operator side, and to unify workflow we are doing it for all reshare/resign/bulk. Do you agree?

Reply:
For my understanding with talking to product I don't agree.
What should happen is:
1. Using code and tools not in this repository the user creates a message and signs it (whether multisig or not)
2. With the DKG he recreates the same reshare message and then sends it together with the signature in one phase

Btw, this process can be improved (just talked with Jordan)...
Instead of recreating the same reshare message you can just input a string of the serialized message and signature as cli parameter.
Apparently the external tool is not finalized so you can come up with whatever serialization schema you want, get it approved with Jordan or Massimo, and post it on the Github issue...

This is better than the original design (my points 1 and 2) because there is a chance we create different messages then the tool.